core-kotlin/src/main/java/com/amplifyframework/kotlin/datastore/KotlinDataStoreFacade.kt

```diff
@@ -170,4 +176,38 @@ class KotlinDataStoreFacade(private val delegate: Delegate = Amplify.DataStore)
             )
         }
     }
+
+    internal class Observation<T>(
+        internal val starts: MutableSharedFlow<Cancelable> = MutableSharedFlow(1),
+        internal val changes: MutableSharedFlow<T> = MutableSharedFlow(1),
+        internal val failures: MutableSharedFlow<DataStoreException> = MutableSharedFlow(1),
+        internal val completions: MutableSharedFlow<Unit> = MutableSharedFlow(1)
+    ) {
+        @Suppress("UNCHECKED_CAST")
+        @FlowPreview
+        internal suspend fun waitForStart(): Flow<T> {
```

Comment:
This is super awesome! 😎

Reply:
Thanks! Appreciate you Insisting on High Standards here. With our back and forth on these `Flow` APIs, I do feel like we ultimately ended up with a better result for the customer.